lib/containers/common.pm

```diff
@@ -130,4 +159,74 @@ sub test_container_image {
     assert_script_run("grep \"Heartbeat from $image\" '/var/tmp/container_$container'");
 }
 
+# Testing openSUSE based images
+sub test_opensuse_based_image {
+    my $image = $_[0];
+    my ($name, $tag) = split(/:/, $image);
+    $tag = 'latest' if (!defined($tag));
+
+    my $runtime = $_[1] //= "docker";
+
+    my $distri  = $_[2] //= get_required_var("DISTRI");
+    my $version = $_[3] //= get_required_var("VERSION");
+
+    assert_script_run("$runtime image pull $image", timeout => 300);
+    assert_script_run("$runtime image ls | grep '$name' | grep '$tag'");
+    # Running executables works
+    assert_script_run qq{$runtime container run --entrypoint '/bin/bash' --rm $image -c 'echo "I work" | grep "I work"'};
+    # It is the right version
+    if ($distri eq 'sle') {
+        my $pretty_version = $version =~ s/-SP/ SP/r;
+        validate_script_output("$runtime container run --entrypoint '/bin/bash' --rm $image -c 'cat /etc/os-release'", sub { /PRETTY_NAME="SUSE Linux Enterprise Server $pretty_version"/ });
+
+        if (is_sle('=12-SP3', $version)) {
+            my $plugin = '/usr/lib/zypp/plugins/services/container-suseconnect';
+            assert_script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin'";
+        } else {
+            my $plugin = '/usr/lib/zypp/plugins/services/container-suseconnect-zypp';
+            assert_script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin -v'";
+            script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin lp'", 420;
+            script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin lm'", 420;
+        }
+    } else {
+        validate_script_output qq{$runtime container run --rm $image cat /etc/os-release}, sub { /PRETTY_NAME="openSUSE (Leap )?${version}.*"/ };
+    }
+    # zypper lr
+    assert_script_run("$runtime run --rm $image zypper lr -s", 120);
+    # zypper ref
+    assert_script_run("$runtime run --name refreshed $image sh -c 'zypper -v ref | grep \"All repositories have been refreshed\"'", 120);
+    # Commit the image
+    assert_script_run("$runtime commit refreshed refreshed-image", 120);
+    # Remove it
+    assert_script_run("$runtime rm --force refreshed", 120);
```

Comment:
Do we need `--force` here? It might swallow some errors when running the `rm` subcommand that we would otherwise caught.

Reply:
```
docker container run --entrypoint '/bin/bash' --rm alpine -c 'echo "I work" | grep "I work"'; echo dw4lt-$?-
docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"/bin/bash\": stat /bin/bash: no such file or directory": unknown.
[31mERRO[0m[0000] error waiting for container: context canceled 
dw4lt-127-
```